File: melodify-backend/src/main/java/com/melodify/entity/MusicAsset.java
package com.melodify.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("music_asset")
public class MusicAsset {

	@TableId(type = IdType.AUTO)
	private Long id;

	/**
	 * 资产编号，对应列 asset_id
	 */
	private String assetId;

	private Long taskId;

	private Long userId;

	private String title;

	private String fileUrl;

	private String coverUrl;

	private Integer durationSec;

	private String format;

	private Integer bitrateKbps;

	private Integer isPublic;

	private String licenseType;

	private Integer status;

	@TableField(fill = FieldFill.INSERT)
	private LocalDateTime createTime;

	@TableField(fill = FieldFill.INSERT_UPDATE)
	private LocalDateTime updateTime;
}

File: melodify-client/src/types/musicAsset.ts
/** 与 `/api/client/music-assets` JSON 对齐 */
import type { MusicTask } from './musicTask'

export interface MusicAsset {
  id: number
  assetId: string
  taskId: number
  userId: number
  title?: string | null
  fileUrl: string
  coverUrl?: string | null
  durationSec?: number | null
  format?: string | null
  bitrateKbps?: number | null
  isPublic?: number | null
  licenseType?: string | null
  status?: number | null
  createTime?: string | null
  updateTime?: string | null
}

export interface MusicAssetDetail {
  asset: MusicAsset
  task?: MusicTask | null
  likeCount: number
  liked: boolean
  prompt?: string | null
  params?: Record<string, unknown> | null
  createTime?: string | null
}
